app/javascript/packages/address-search/hooks/use-usps-locations.ts

```diff
@@ -0,0 +1,108 @@
+import { useState, useRef, useEffect, useCallback } from 'react';
+import { request } from '@18f/identity-request';
+import { t } from '@18f/identity-i18n';
+import useSWR from 'swr/immutable';
+import type { Location, FormattedLocation, LocationQuery, PostOffice } from '../types';
+import { formatLocations, snakeCase, transformKeys } from '../utils';
+
+const requestUspsLocations = async ({
+  locationsURL,
+  address,
+}: {
+  locationsURL: string;
+  address: LocationQuery;
+}): Promise<FormattedLocation[]> => {
+  const response = await request<PostOffice[]>(locationsURL, {
+    method: 'post',
+    json: { address: transformKeys(address, snakeCase) },
+  });
+
+  return formatLocations(response);
+};
+
+function requestAddressCandidates({
+  unvalidatedAddressInput,
+  addressSearchURL,
+}: {
+  unvalidatedAddressInput: string;
+  addressSearchURL: string;
+}): Promise<Location[]> {
+  return request<Location[]>(addressSearchURL, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    json: { address: unvalidatedAddressInput },
+  });
+}
+
+export default function useUspsLocations({
```

Comment:
This function seems to have an implicit dependency on an address input? I feel like now that it's exported out to a hook it would be good to make it clear what it depends on, so like


```
const { ... } = useUspsLocations({ addressInput ... })
```
right?